### PR TITLE
Improve controller report xls.

### PR DESCRIPTION
**Pull request by feketemihai**
I don't have more time to allocate on this, so i only added usual styles defined in odoo templates, probably when i will have time i will update them and to create a function to transform div classes to tables.

### CODE REVIEW

**Comment by nhomar** [comment]
Interesting.

Can you share one module where you are using this approach?, 

because some of our reports depend of it and with this improvements we need to be sure any of the tests are broken (yours or ours) then if you share how are you using your approach we can help to finish the tests or may be improve the logic.

Regards.

**Comment by nhomar** [comment]
You can see a lot of lints failing also, may be it can help to improve your own internal use:

https://travis-ci.org/Vauxoo/addons-vauxoo/jobs/123819734

**Comment by feketemihai** [comment]
Hy @nhomar , i am fixing now the pylint and pep8 issues, i am using this with https://github.com/feketemihai/odoo_reports_xls/tree/master/account_financial_report_xls, the trial balance is extended from https://github.com/odoo-romania/l10n-romania/tree/8.0/l10n_ro_account_report, the rest and from odoo v8 base.

**Comment by nhomar** [comment]
Nice.!

I just read quickly your report and I recommends to you use ours (and improve what is blocking your needs) we use it for something like 6 differents countries and it is pretty generic.

https://github.com/Vauxoo/odoo-afr/tree/8.0/account_financial_report
https://github.com/Vauxoo/odoo-ifrs

I hope it helps you and on that way we can merge both concepts toghether.... this controller was don to extend quicckly any report already done, and I see you cut the scope to work yours purelly to xls (AFAICU).

Then In that way we can help you.

Conceptually what you did here look pretty cool to me, but that report engine is KEY in our internal deployments I hope you can use them.

**Comment by feketemihai** [comment]
@Nhomar, it should not modify anything, untill these modifications, there were no style defined on the reports when export to excel, the module only brings beside table_header and table_body the thead and tbody and try to interpreter the html with style and adjust rowspan, colspan to set proper width and cell formating. I am testing now with ifrs and afr but with demo data.

**Comment by nhomar** [comment]
@feketemihai Yes absollutly.

Our approach is simple has a simple way to export **as is** the report to excel.

But I am agree with your approach, just to explain the important part:
- **priority 1**: The results.
- **priority 2**: Plain Excel exported to quick analysis.
- **priority 3**: Plain text with more than simply plain number (your approach).

Then, your approach can not brake the **priority 2**.

But you are right and it was intentional, but both must work toghether tested, that's important because we have those reports deployed into production and I am eager to merge your approach, I think is useful (and it was in our roadmap) but for a long future, we try to render everything fancy just in the PDF.

Then I hope it helps you.

PS: When I mentioned try our approaches on accounting I mean the **accounting part**.... Your approach is a good help for this featue but I know they will not fit the fancy part (**priority 3**)

**Comment by feketemihai** [comment]
[ifrs_report.ifrs_portrait_pdf_report.pdf](https://github.com/Vauxoo/addons-vauxoo/files/227414/ifrs_report.ifrs_portrait_pdf_report.pdf)
[ifrs_report.ifrs_portrait_pdf_report.xlsx](https://github.com/Vauxoo/addons-vauxoo/files/227415/ifrs_report.ifrs_portrait_pdf_report.xlsx)
I still have a difference on the header, it's getting it twice...but not all, only the first table

<div class="header">
        <table width="100%"><tr><td width="30%">
                  <div>
                    </div>

```
          </td>
          <td width="70%">
              <table style="width: 100%; text-align:center;" id="table_header"><tr data-oe-id="569" data-oe-model="ir.ui.view" data-oe-field="arch" data-oe-xpath="/t[1]/div[1]/table[1]/tr[1]/td[2]/table[1]/tr[1]"><td><div class="td_company_title"><span data-oe-model="res.company" data-oe-id="1" data-oe-field="name" data-oe-type="char" data-oe-expression="o.company_id.name">YourCompany</span></div></td></tr><tr><td>
                          <div class="td_company">
                               ALL PERIODS OF THE FISCALYEAR 

                          </div>
                      </td>
                  </tr><tr data-oe-id="569" data-oe-model="ir.ui.view" data-oe-field="arch" data-oe-xpath="/t[1]/div[1]/table[1]/tr[1]/td[2]/table[1]/tr[3]"><td><div class="td_company"><span data-oe-model="account.fiscalyear" data-oe-id="1" data-oe-field="name" data-oe-type="char" data-oe-expression="o.fiscalyear_id.name">Fiscal Year X 2016</span></div></td></tr></table></td>
      </tr></table><table class="basic_table" width="100%" border="0"><thead><tr><th class="celdaTituloTabla" width="100%">
                        <span data-oe-model="ifrs.ifrs" data-oe-id="8" data-oe-field="name" data-oe-type="char" data-oe-expression="o.name">DEMO IFRS REPORT - VX ACCOUNTS</span> (Expressed in <span>EUR</span> @ <span>04/20/2016</span>)
                    </th>

            </tr></thead></table></div>
```

**Comment by nhomar** [comment]
@hbto Your opinion is welcome here.

Thanks.

File: controller_report_xls/controllers/xfstyle.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    "Wheat": "#F5DEB3",
+    "White": "#FFFFFF",
+    "WhiteSmoke": "#F5F5F5",
+    "Yellow": "#FFFF00",
+    "YellowGreen": "#9ACD32"
+}
+
+# Culled from a table at http://www.mvps.org/dmcritchie/excel/colors.htm
+XLWT_COLORS = [
+    (0, 0, 0), (255, 255, 255), (255, 0, 0), (0, 255, 0), (0, 0, 255),
```
Comment (nhomar) [inline controller_report_xls/controllers/xfstyle.py]: 1 line per element is better for future diffs.

Comment (feketemihai) [inline controller_report_xls/controllers/xfstyle.py]: I will add it in the next commit, but i am waiting for other comments, to fix all in one...

**Comment by hbto** [comment]
I like the approach being taken by @feketemihai 

while the end result is still for xls a report with the Data Row and Cols,
I think we can endure some repetetions of headers at the top
or footers at the bottom, though data the most important thing
shall remain non-repeated.

Best Regards

**Comment by feketemihai** [comment]
@nhomar Can you please review it again...Thanks.

[ifrs_report.ifrs_landscape_pdf_report.pdf](https://github.com/Vauxoo/addons-vauxoo/files/233899/ifrs_report.ifrs_landscape_pdf_report.pdf)
[ifrs_report.ifrs_landscape_pdf_report.xlsx](https://github.com/Vauxoo/addons-vauxoo/files/233900/ifrs_report.ifrs_landscape_pdf_report.xlsx)

**Comment by nhomar** [comment]
@feketemihai 

I opened the Excel File into LibreOffice 5.1.2.2 and I see this:



Numbers are Perfect but colors does not fit with the PDF file, is that what you expect?

Can you share a PrintScreen with your result into Excel (Version).

It looks good to me (no final code review yet)

Thanks A LOT for the contribution BTW.

**Comment by feketemihai** [comment]
@nhomar It's the same for me, because on the code there is a code for transofrmation into rgb and take the closer one available, the #dcdcdc color has a rgb of (220, 220, 220), matched with excel colors by distance is #CCCCFF which is (204, 204, 255)...

Please see https://github.com/python-excel/xlwt/blob/master/xlwt/Style.py#L376

**Comment by nhomar** [comment]
@feketemihai Ok, I just want to know it was the expected behavior.
@hbto before the merge/rebase can you give it a last view and maybe check in a customer environment to ensure all is Ok!.

Regards.

**Comment by hbto** [comment]
@nhomar I will be reviewing on a customer's enviroment.
@feketemihai Thanks for this great feature coming to this module.

Regards

**Comment by hbto** [comment]
Hi @feketemihai 
As we are testing your PR with our CI Server on Our Customer Project
it happes that is is required that you add cssutils python dependency on this file

https://github.com/feketemihai/addons-vauxoo/blob/8.0/requirements.txt

in the meantime I have added it (temporary) on Customer's Project 
But permanent locations is in the previous file I have sent you.

Test will go on, though it is required that you add that to merge the PR.

Regards.

**Comment by feketemihai** [comment]
@nhomar @hbto I will rebase after the translation files are added...

**Comment by hbto** [comment]
@feketemihai 

I have tested with aging_due_report module from addons-vauxoo,
this module depends on regarding controller, 
and it is not working

[aging_due_report.aging_due_report_qweb.xlsx](https://github.com/Vauxoo/addons-vauxoo/files/236874/aging_due_report.aging_due_report_qweb.xlsx)

In the case of odoo-afr/account_financial_report

It is working like expected so far.

Best Regards

**Comment by nhomar** [comment]
@feketemihai The last step is the rebase, just waiting your input here.

**Comment by feketemihai** [comment]
@nhomar I will do it in the morning...now it's 9 pm in romania...

**Comment by feketemihai** [comment]
@nhomar @hbto I've updated code for getting the horizontal and vertical alignment, but i'm having issues parsign style defined in aging_due_report, please see comparition of 
https://github.com/Vauxoo/addons-vauxoo/blob/8.0/aging_due_report/data/aging_due_report_style.xml

and 

https://github.com/Vauxoo/odoo-ifrs/blob/8.0/ifrs_report/report/layouts.xml

The problem is i don't know how exactly to retrieve the style defined in aging module, it's the same style defined for table, overwritten for columns, alternate rows, use of first-child, last-child..
Now the report is printed, but the style interpreter is not getting the correct style defined in text/css, now  from style defined:

.table_title_bg_color {
  border-collapse: collapse; }
  .table_title_bg_color > thead > tr > th,
  .table_title_bg_color > tbody > tr > th,
  .table_title_bg_color > tfoot > tr > th {
    border-bottom: 1px solid #DFDFDE;
    border-top: 1px solid #DFDFDE;
    background-color: #A41D35;
    font-size: 12;
    color: white; }

it only gets 

.table_title_bg_color {
  border-collapse: collapse; }

**Comment by hbto** [comment]
Thanks for your reply @feketemihai 

If you feel compeled to do so, I encourage you to use other styles in aging module
that can have the similar outcomes and works with your changes

Regards

**Comment by feketemihai** [comment]
@hbto I will try to do it, but there are other modules that use this style definition, untill now i found user_story and crm_claim_summary_report...

**Comment by nhomar** [comment]
Thanks for the review, if you simply share a list we can tell you which one use this feature, because if them do not use the export xls feature it does not make much sense migrate them.

Really thanks for the review.

**Comment by feketemihai** [comment]
@nhomar i only found those 2 modules, user_story and crm_claim_summary_report, which have style declared as the one from aging module...

**Comment by nhomar** [comment]
Do not worry, those module do not use that feature the important one in the aging.

BTW, we can merge if simple you set in the README a message mentioning the limitation, then we simply plan the fix afterwards for others.

**Comment by nhomar** [comment]
@hugho-ad I was waiting this module has a config no a ir.config.parameter.

**Comment by feketemihai** [comment]
@nhomar I will update the style in aging now...and add a remark in readme after...about the style definition

**Comment by feketemihai** [comment]
@nhomar @hbto Could you check if it's ok for you...
[aging_due_report.aging_due_report_qweb.pdf](https://github.com/Vauxoo/addons-vauxoo/files/238330/aging_due_report.aging_due_report_qweb.pdf)
[aging_due_report.aging_due_report_qweb.xlsx](https://github.com/Vauxoo/addons-vauxoo/files/238331/aging_due_report.aging_due_report_qweb.xlsx)

**Comment by nhomar** [comment]
The only point i worried about is that the number formating in the xls is Integer and the PDF is float, is that expected? all other points looks pretty good to me.

**Comment by feketemihai** [comment]
@nhomar All the invoices were integers, i've updated a invoice with decimals, and set a Separator Format to [3,3,3,3], and the result is 
[aging_due_report.aging_due_report_qweb.xlsx](https://github.com/Vauxoo/addons-vauxoo/files/238361/aging_due_report.aging_due_report_qweb.xlsx)
 even that the cell format in excell remains to general...calculation of amount are ok in excel...selecting multiple cell gives you the amount...

**Comment by nhomar** [comment]
Ok, excellent, then good for me.

File: controller_report_xls/controllers/main.py [hunk 2]
```diff
@@ -1,45 +1,174 @@
...
+from ..controllers.xfstyle import css2excel
 
 from openerp.addons.report.controllers import main
 from openerp.addons.web.http import route, request  # pylint: disable=F0401
 from werkzeug import url_decode  # pylint: disable=E0611
 import simplejson
-import lxml.html
+from lxml import etree
+import cssutils
+from cssutils import parseString
```
Comment (nhomar) [inline controller_report_xls/controllers/main.py]: Can you import this inside a try please.

As it is not a normal dependency declare this repository bring unexpected traceback with just run the server with this depclared.

more information:

https://yelizariev.github.io/odoo/development/2015/07/06/external-dependencies.html

**Comment by feketemihai** [comment]
@nhomar i will deal with aging pylint issues tomorow...

**Comment by feketemihai** [comment]
@nhomar Can you enlight me with pylint issues, it has remained only the reports name and _columns???

**Comment by nhomar** [comment]
Ignore them, those are the validation to migrate to new api any module we touch from now in advance. And only check the module you touch ion the PR itself, as you made some improvements on that module it is triggering the checks, ignore them, I think they are out of your scope on this PR but that's your decision, just let me know.

**Comment by feketemihai** [comment]
@nhomar I will upgrade now...

**Comment by feketemihai** [comment]
@hbto Can you please recheck if it's ok for you..Thanks..

**Comment by hbto** [comment]
@feketemihai Thanks a lot. I will be checking for it

Regards.

**Comment by hbto** [comment]
@feketemihai 

In order to keep you posted 
We are testing your PR against one of our customer's instance
We are having some troubles with real data, thought with demo 
data everything seems to be ok.

When we reach any meaningful results that we can provide you we will
let you know.

Best Regards

**Comment by feketemihai** [comment]
@hbto Ok..thanks...please tell me what if the issues are from the code or non matched styling tags...
@nhomar Is it ok to put myself in the authors tab of the module?

**Comment by nhomar** [comment]
On Tue, May 3, 2016 at 10:51 PM, Fekete Mihai notifications@github.com
wrote:

> @nhomar https://github.com/nhomar Is it ok to put myself in the authors
> tab of the module?
> 
> I prefer a contributors section in the README, Are you ok with that?.
> (describe your contributions there, pick where you think is better for you
> in the README). And it gives more presence in the apps store that the
> author list.

Remember update the README and de description file.

## 

Nhomar Hernandez
CEO Vauxoo.
Site: http://vauxoo.com
Twitter: @nhomar
Blog: http://nhomar.com
Github User: https://github.com/nhomar
Odoo Gold Partner
Skype: nhomar00 (Envia mail previo no lo superviso siempre).
HangOut: nhomar@vauxoo.com
Móvil Venezuela:
+58 4144110269
Móvil México:
+52 1 4773933942

**Comment by feketemihai** [comment]
@nhomar Updated files, is it ok like this?

**Comment by nhomar** [comment]
I recommend this format:

```
"Your Name <you@email.com>" <<<<--With brackets
```

**Comment by feketemihai** [comment]
I wanted to be the same with others Contributors tabs on the repo...
https://github.com/Vauxoo/addons-vauxoo/blob/c311ef248340d3f1f9bc699aa61bfc44dc443efb/base_vat_country/README.rst

**Comment by nhomar** [comment]
Yes perfect.

I just said that the email is missing in your name, you can add it into both files descriptions and readme.

**Comment by moylop260** [comment]
@hbto 
Please, Could you review this changes with real data?

File: aging_due_report/wizard/wizard.py [hunk 3]
```diff
@@ -436,18 +314,24 @@ def _get_lines_by_partner_without_invoice(
...
+                              ['partner_id', 'reconcile_partial_id', 'debit',
                                'credit', 'amount_currency', 'currency_id'],
-            load=None, context=context)
+                              load=None)
+        try:
+            from pandas import DataFrame
+        except ImportError:
+            _logger.info('aging_due_report is declared '
+                         ' from addons-vauxoo '
+                         ' you will need: sudo pip install pandas')
```
Comment (nhomar) [inline aging_due_report/wizard/wizard.py]: can you import this outside the def please.

**Comment by hbto** [comment]
@moylop260 Yes I will be checking PR in a real data,
as suggested in the other PR https://github.com/Vauxoo/addons-vauxoo/pull/926
I will first go with mccabe, 
having that properly set I will continue with the other one

Regards.

**Comment by feketemihai** [comment]
@moylop260 Can you help me with the pylint issues...if i move dataframe import at top of file as @nhomar suggested...

**Comment by moylop260** [comment]
@feketemihai 
The `wrong-import-position` shouldn't showed in `try-import-except` case.
I will discuss this case with original pylint project.
You could ignore it for now.

**Comment by moylop260** [comment]
<s>For `too-many-boolean-expressions` we can disabled it for this case for now

``` patch
    @api.model
    def _get_amount_span(self, field_names, res_line, doc, spans, direction):
+        # pylint: disable=too-many-boolean-expressions
        len_span = len(spans) if spans else 0
```

<s/>

**Comment by moylop260** [comment]
For `too-many-boolean-expressions` I will fix it with the following change:

``` patch
diff --git aging_due_report/wizard/wizard.py aging_due_report/wizard/wizard.py
index 5f5a438..f7a71d8 100644
--- aging_due_report/wizard/wizard.py
+++ aging_due_report/wizard/wizard.py
@@ -100,19 +100,22 @@ class AccountAgingWizardPartner(osv.osv_memory):
             span = 'span%02d' % (item + 1)
             if span not in field_names:
                 continue
+            is_first_item = first_item == item
+            is_last_item = last_item == item
+            is_mid_item = not is_first_item and not is_last_item
             if not direction:
-                if first_item == item and doc.due_days <= 0 or \
-                        last_item == item and doc.due_days <= spans[item] or \
-                        doc.due_days <= spans[item] and \
-                        doc.due_days > spans[item + 1]:
-                    res_line[span] += doc.residual
+                first_span = is_first_item and doc.due_days <= 0
+                last_span = is_last_item and doc.due_days <= spans[item]
+                mid_span = is_mid_item and doc.due_days <= spans[item] and \
+                    doc.due_days > spans[item + 1]
             else:
-                if first_item == item and doc.due_days > 0 \
-                        and doc.due_days <= spans[item] or \
-                        last_item == item and doc.due_days > spans[item] or \
-                        doc.due_days > spans[item] and \
-                        doc.due_days <= spans[item + 1]:
-                    res_line[span] += doc.residual
+                first_span = is_first_item and doc.due_days > 0 \
+                    and doc.due_days <= spans[item]
+                last_span = is_last_item and doc.due_days > spans[item]
+                mid_span = is_mid_item and doc.due_days > spans[item] and \
+                    doc.due_days <= spans[item + 1]
+            res_line[span] += (first_span or last_span or mid_span) \
+                and doc.residual or 0

     def _get_amount(self, cr, uid, ids, field_names, arg, context=None):
         if context is None:

```

I don't want to change it to avoid a code conflict.
I will wait your merge to change it.

Feel free to apply it

**Comment by moylop260** [comment]
FYI I updated my https://github.com/Vauxoo/addons-vauxoo/pull/860#issuecomment-218773740

**Comment by feketemihai** [comment]
@moylop260 i'm out of office today...i will update on monday. Thanks.

**Comment by moylop260** [comment]
The travis build show a error by redundant id and wrong-import-position
I can fix it after merge

Then 👍

**Comment by moylop260** [comment]
thanks @feketemihai

File: controller_report_xls/controllers/main.py [hunk 4]
```diff
@@ -1,45 +1,200 @@
...
+def get_css_style(csstext, style):
+    cssstyle = ""
+    if csstext:
+        try:
+            import cssutils
+            from cssutils import parseString
+            cssutils.log.setLevel(logging.CRITICAL)
+        except ImportError:
+            pass
+        cssnode = parseString(csstext)
```
Comment (moylop260) [inline controller_report_xls/controllers/main.py:50]: Show `undefined parseString` if cssutils is not installed, right?

**Comment by hbto** [comment]
@feketemihai @nhomar @moylop260 

Should proceed with a testing on Customer's Instance 
to check for workability in Production environment?

Regards

**Comment by nhomar** [comment]
On Mon, May 16, 2016 at 7:35 PM, Hbto [ Vauxoo ] notifications@github.com
wrote:

> to check for workability in Production environment?
> 
> Yes pliz... in order to merge....

## 

Nhomar Hernandez
CEO Vauxoo.
Site: http://vauxoo.com
Twitter: @nhomar
Blog: http://nhomar.com
Github User: https://github.com/nhomar
Odoo Gold Partner
Skype: nhomar00 (Envia mail previo no lo superviso siempre).
HangOut: nhomar@vauxoo.com
Móvil Venezuela:
+58 4144110269
Móvil México:
+52 1 4773933942

File: controller_report_xls/controllers/main.py (hunk 4 above)
Comment (feketemihai) [inline controller_report_xls/controllers/main.py:50]: yes...if cssutils not installed...do you think i should log if cssutils are not installed...?

Comment (moylop260) [inline controller_report_xls/controllers/main.py:50]: Maybe return a empty value early

**Comment by hbto** [comment]
Hello @feketemihai 

I have tested your code in a Testing Environment with Real Data,
I have found that this change is needed to be made in order to
ensure everything works fine.

```
diff --git a/aging_due_report/wizard/wizard.py b/aging_due_report/wizard/wizard.py
index 3170baf..ee2659f 100644
--- a/aging_due_report/wizard/wizard.py
+++ b/aging_due_report/wizard/wizard.py
@@ -363,7 +363,7 @@ class AccountAgingPartnerWizard(models.TransientModel):
                     if amu_brw.date == date_emission])

                 doc = {
-                    'partner_id': partner_id,
+                    'partner_id': partner_id[0],
                     'aml_id': aml_id[0],
                     # 'wh_vat': wh_vat,
                     # 'wh_islr': wh_islr,
```

@nhomar having added this excerpt of code PR will be ready to be merge

Best Regards

**Comment by moylop260** [comment]
@hbto 
Thank you
I'll merge with squash manually and I'll add your change

**Comment by moylop260** [comment]
Merged 85dd227cbdfeb70f14d83f1f0b4a0cc3f53a960c

**Comment by moylop260** [comment]
@feketemihai thanks for your excellent contribution!

**Comment by feketemihai** [comment]
@moylop260 Thanks and sorry that i was busy with other stuff and wasn't able to finish @hbto comment...

**Comment by hbto** [comment]
@feketemihai 

Hello, thanks for your effort bringing a new look and feel on the module,
and improving the code in the controller.

I would like to know how are you implementing this controller in your modules

Regards.

**Comment by feketemihai** [comment]
@moylop260 I already sent to @nhomar , i will use it with a part of https://github.com/feketemihai/odoo_reports_xls but now i will have to adapt the style defined on the qweb and to change the div tables section to tables...but it's done in a hurry just to add the xls export...the idea is that i need an excel export for accounting audit( the reports are already in the https://github.com/odoo-romania/l10n-romania repo), and instead of adding new report that goes directly in excel, it will much easier to maintain only one report (code and view), and export to excel to interpret the report...

**Comment by hbto** [comment]
> i will use it with a part of https://github.com/feketemihai/odoo_reports_xls 

@feketemihai Oh! I can see now, I thought you were already deployed this module in one of your reports.

Best Regards

**Comment by hbto** [comment]
@feketemihai 

> and instead of adding new report that goes directly in excel, it will much easier to maintain only one report (code and view), and export to excel to interpret the report...

Yes that have been the idea behind this controller allowing you write one report-code and having at least two different formats of same report.

Regards

**Comment by feketemihai** [comment]
@hbto It's already deployed with private reports with style from when i started to work on the PR, so it's  more than one month in a multicompany production env, and it's working ok, no bad experience from users...

**Comment by feketemihai** [comment]
@nhomar @moylop260 @hbto Too bad that this module is not in the https://github.com/OCA/reporting-engine,  instead of other modules that adds new reports to excel...

**Comment by feketemihai** [comment]
In a few months i will have to create a new one, based on xlsxwritter to support more than 65k lines...

**Comment by hbto** [comment]
@feketemihai Eventually it will find its way to https://github.com/OCA/reporting-engine in the meantime let it live long and prosper in https://github.com/Vauxoo/addons-vauxoo, 

@nhomar & @moylop260 will provide us with a course of action regarding its handover to another project.

**Comment by hbto** [comment]
@feketemihai 

> In a few months i will have to create a new one, based on xlsxwritter to support more than 65k lines...

That is great!!!

Are you willing to provide that feature in this same module?

That would be awesome!!!

**Comment by hbto** [comment]
@feketemihai How did you find your way into this module `controller_report_xls`

**Comment by feketemihai** [comment]
@hbto Yes..probably to overwrite it, not to have 2 options, xls and xlsx...

**Comment by feketemihai** [comment]
@hbto I am always searching for modules that makes my life easier...i started playing with it for more than 3 months and to do small adjustments to make it more general...and since we are not Odoo partners, we only use Odoo internaly, and i am the only developer on Odoo, it will save me a lot of time designing new reports just for export to excel...it's much easier to add the field in the wizard and in the wizard view...

**Comment by hbto** [comment]
@feketemihai 

Knowing about your good experience, on this module `controller_report_xls`, compels me to encourage you to write a post in the Odoo community mailing list so more people is acknowledged on the existence of this module, that could fast-track arrival of this module onto  https://github.com/OCA/reporting-engine 

Best Regards

**Comment by feketemihai** [comment]
@hbto Initially, i will update the reports for romanian legislation, update style, create a separate repo for excel reports, and let's hope others will use it and join the developments...but i think the style mapping will be improved consistently, there are a lot of features in html style that are not mapped now...so i wait a little bit...

**Comment by feketemihai** [comment]
@nhomar @hbto @moylop260 Changing to xlsxwritter and after to propose it to OCA will be more appropriate, since as far as i seen it has more options than xlwt, and let's face it, who is still using excel 2003??

**Comment by hbto** [comment]
@feketemihai 

> Changing to xlsxwritter and after to propose it to OCA will be more appropriate,

It seems to me appropriate too

Regarding the use of current library, it fulfilled the initial goal but as demand grows
and new challenges appear it comes to show now needs to be fulfilled and new
approaches to be taken as you have well pointed out.

Discuss with @nhomar & @moylop260 the use of the another library 
beside that of more 65K lines provide other benefits that will come with it.

Regards

**Pull request closed without merging**